File: src/baseDatos/Serializador.java
package baseDatos;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.util.List;

import gestorAplicacion.Autocine.Autocine;
import gestorAplicacion.Salas.Puesto;
import gestorAplicacion.Salas.Sala;
import gestorAplicacion.Taquilla.Funcion;
import gestorAplicacion.Salas.Sala3D;
import gestorAplicacion.Taquilla.Pelicula;
import gestorAplicacion.Taquilla.Ticket;
import gestorAplicacion.Persona.Cliente;
import gestorAplicacion.Salas.Sala2D;


public class Serializador {

	public static <E> void serializar(List<E> lista, String className) {
		FileOutputStream fileOut;

		try {
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + className + ".txt";
			// se crea un fileoutputstream para saber donde serializar los archivos
			fileOut = new FileOutputStream(path);
			// Se crea un objeto output stream para poder escribir en el archivo
			ObjectOutputStream out = new ObjectOutputStream(fileOut);
			// Guardamos la lista de objetos
			out.writeObject(lista);
			out.close();
			fileOut.close();
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}

	/**
	 * Serializamos todas las clases que necesitamos
	 */
	public static void serializarTodo() {
		Serializador.serializar(Autocine.getAutocine(), "Autocine");
		Serializador.serializar(Funcion.getFunciones(), "Funciones");
		Serializador.serializar(Sala3D.getSalas3D(), "Salas 3D");
		Serializador.serializar(Sala.getSalas(), "Salas");
		Serializador.serializar(Pelicula.getPeliculas(), "Peliculas");
		Serializador.serializar(Puesto.getPuestos(), "Puestos");
		Serializador.serializar(Cliente.getClientes(), "Clientes");
		Serializador.serializar(Ticket.getTicket(), "Tickets");
		Serializador.serializar(Sala2D.getSalas2D(), "Salas 2D");
	}

	public static void serializar(Autocine autocine) {
		// TODO Auto-generated method stub
		
	}
}

File: Practica 2/AutoCine/src/baseDatos/serializador.py
import pickle
from gestorAplicacion.Autocine.Autocine import Autocine
from gestorAplicacion.Persona.Cliente import Cliente
from gestorAplicacion.Salas.Puesto import Puesto
from gestorAplicacion.Salas.Sala import Sala
from gestorAplicacion.Salas.Sala2D import Sala2D
from gestorAplicacion.Salas.Sala3D import Sala3D
from gestorAplicacion.Taquilla.Funcion import Funcion
from gestorAplicacion.Taquilla.Pelicula import Pelicula
from gestorAplicacion.Taquilla.Ticket import Ticket
import pathlib
import os

class Serializador():
    def serializar(lista, className):
        def camino(className):
            string = os.path.join(pathlib.Path(__file__).parent.absolute(), "temp\\"+className+".txt")
            return string
        try:
            # Creo el archivo pickle para guardar los objetos.
            picklefile = open(camino(className), 'wb')
            # Pickle el objeto en el archivo.
            pickle.dump(lista, picklefile)
            # Cierro el archivo para guardar.
            picklefile.close()
            
        except:
            print("paila tuqui tuqui muñeco")

    def serializarTodo():

        Serializador.serializar(Autocine.getAutocine(), "Autocine")
        Serializador.serializar(Cliente.getClientes(), "Clientes")
        Serializador.serializar(Puesto.getPuestos(), "Puestos")
        Serializador.serializar(Sala.getSalas(), "Salas")
        Serializador.serializar(Sala2D.getSalas2D, "Salas2D")
        Serializador.serializar(Sala3D.getSalas3D, "Salas3D")
        Serializador.serializar(Funcion.getFunciones(), "Funciones")
        Serializador.serializar(Pelicula.getPeliculas(), "Peliculas")
        Serializador.serializar(Ticket.getTickets(), "Tickets")